packages/fxa-auth-server/scripts/stripe-products-and-plans-to-firestore-documents/stripe-product-and-plans-converter.ts

```diff
@@ -0,0 +1,344 @@
+    )) {
+      capabilities[capability] = commaSeparatedListToArray(
+        product.metadata[capability]
+      );
+    }
+    return capabilities;
+  }
+
+  // TODO JSDoc
+  stylesMetadataToStyleConfig(product: Stripe.Product): StyleConfig {
+    const styleConfig: StyleConfig = {};
+    for (const key of StyleConfigKeys) {
+      const value = product.metadata[key];
+      if (value) {
+        styleConfig[key] = value;
+      }
+    }
+    return styleConfig;
+  }
+
+  // TODO JSDoc
+  supportMetadataToSupportConfig(
+    stripeObject: Stripe.Product | Stripe.Plan
+  ): SupportConfig {
+    const supportConfig: SupportConfig = {};
+    if (!stripeObject.metadata) {
+      return supportConfig;
+    }
+    for (const key of SupportConfigKeys) {
+      const metadataKeyPrefix = `support:${key}`;
+      const metadataValues = this.getArrayOfStringsFromMetadataKeys(
+        stripeObject.metadata,
+        metadataKeyPrefix
+      );
+      if (metadataValues.length > 0) {
+        supportConfig[key] = metadataValues;
+      }
+    }
+    return supportConfig;
+  }
+
+  // TODO JSDoc
+  uiContentMetadataToUiContentConfig(
+    stripeObject: Stripe.Product | Stripe.Plan
+  ): UiContentConfig {
+    const uiContentConfig: UiContentConfig = {};
+    if (!stripeObject.metadata) {
+      return uiContentConfig;
+    }
+    for (const key of UiContentConfigKeys) {
+      let value;
+      // TODO: How to filter keys by type (e.g. `string` versus `string[]`);
+      // should `details` have a sub-subtype? Would that even help?
+      if (key === 'details') {
+        const metadataKeyPrefix = `product:${key}`;
+        value = this.getArrayOfStringsFromMetadataKeys(
+          stripeObject.metadata,
+          metadataKeyPrefix
+        );
+        if (value.length > 0) {
+          uiContentConfig[key] = value;
+        }
+        continue;
+      }
+      value =
+        stripeObject.metadata[key] ?? stripeObject.metadata[`product:${key}`];
+      if (value) {
+        uiContentConfig[key] = value;
+      }
+    }
+    return uiContentConfig;
+  }
+
+  // TODO JSDoc
+  urlMetadataToUrlConfig(
+    stripeObject: Stripe.Product | Stripe.Plan
+  ): UrlConfig {
+    const urlConfig: UrlConfig = {};
+    if (!stripeObject.metadata) {
+      return urlConfig;
+    }
+    for (const key of UrlConfigKeys) {
+      const value =
+        stripeObject.metadata[`${key}URL`] ??
+        stripeObject.metadata[`${key}Link`] ??
+        stripeObject.metadata[`product:${key}URL`];
+      if (value) {
+        urlConfig[key] = value;
+      }
+    }
+    return urlConfig;
+  }
+
+  // TODO JSDoc
+  // This method will return a valid ProductConfig, but it will need to be
+  // appended with any locale data for Stripe plans if present.
+  stripeProductToProductConfig(product: Stripe.Product): ProductConfig {
+    // Firestore cannot serialize class instances
+    const productConfig: any = {};
+
+    // BaseConfig
+    productConfig.active = product.active;
```

Comment:
Seeing this makes me question what my original intention was with it.  I'm certain it's not to duplicate the value from the Stripe product/plan; Stripe should remain the only source of truth for that.

I think I wanted it as a flag to determine whether to load the config, but that doesn't really gets us much...if the Stripe plan is active but the config isn't, then how would we resolve that?

Let's delete it??

Reply:
It seems to me the concept behind active/inactive is a good one to keep for our more abstract `productConfig` and `planConfig` representations. E.g. what if a PM wants to discontinue a plan and block any new subscriptions? How would that be indicated?

Maybe we can set it to `true` here for now, and it can be toggled via the config UI in the future?